### PR TITLE
Run3-sim152Z Take care of difference between DDD and DD4hep in SteppingAction classes (replace the PR #43499)

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Take care of the difference between DDD and DD4hep in SteppingAction classes (replace the PR #43499)

#### PR validation:

Use the runTheMatrix test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4Core/Application/src/Phase2SteppingAction.cc [hunk 1]
```diff
@@ -13,6 +13,8 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/Utilities/interface/isFinite.h"
 
+#include <DD4hep/Filter.h>
```
Comment (civanch) [inline SimG4Core/Application/src/Phase2SteppingAction.cc:16]: @bsunanda , let us somehow remove this header.

File: SimG4Core/Application/src/SteppingAction.cc [hunk 2]
```diff
@@ -13,7 +13,9 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/Utilities/interface/isFinite.h"
 
-//#define DebugLog
+#include <DD4hep/Filter.h>
```
Comment (civanch) [inline SimG4Core/Application/src/SteppingAction.cc:16]: The same here, better do not depend on external package - std should have methods to resolve such problems.

File: SimG4Core/Application/interface/Phase2SteppingAction.h [hunk 3]
```diff
@@ -64,6 +64,7 @@ class Phase2SteppingAction : public G4UserSteppingAction {
   std::vector<const G4Region*> deadRegions;
   std::vector<G4LogicalVolume*> ekinVolumes;
   std::vector<int> ekinPDG;
+  G4String cmseName_, trackerName_, caloName_, btlName_;
```
Comment (civanch) [inline SimG4Core/Application/interface/Phase2SteppingAction.h:67]: These variables are needed only for initialisation. In run time optimal would be to use G4String, but at initialisation better to use std::string, which would be easier to connect to other std methods.

File: SimG4Core/Application/src/Phase2SteppingAction.cc (hunk 1 above)
Comment (bsunanda) [reply SimG4Core/Application/src/Phase2SteppingAction.cc:16]: I am working on this

File: SimG4Core/Application/interface/Phase2SteppingAction.h (hunk 3 above)
Comment (bsunanda) [reply SimG4Core/Application/interface/Phase2SteppingAction.h:67]: I shall do that when I remove dd4hep dependence

**Comment by bsunanda** [comment]
@civanch I am working on something which will take care of all your comments later. Please approve this and let it merge

**Comment by civanch** [comment]
+1

let us merge this and and open an issue to take out dependence on DD4hep

**Pull request merged**